dace/codegen/targets/unroller.py

```diff
@@ -23,31 +30,86 @@ def __init__(self, frame_codegen: DaCeCodeGenerator, sdfg: dace.SDFG):
+        def nsdfg_prepare_unroll(scope, paramname, paramval):
+            backup = []
+            for node in scope.nodes():
+                if (isinstance(node, nd.NestedSDFG)):
+                    backup.append((node, node.unique_name,
+                                   node.sdfg.name,
+                                   node.symbol_mapping,
+                                   node.sdfg.constants_prop))
+                    node.unique_name = copy.deepcopy(node.unique_name)
+                    node.sdfg.name = copy.deepcopy(node.sdfg.name)
+                    node.symbol_mapping = copy.deepcopy(node.symbol_mapping)
+                    node.sdfg.constants_prop = copy.deepcopy(node.sdfg.constants_prop)
+                    node.unique_name = f"{node.unique_name}_{param}{paramval}"
+                    node.sdfg.name = f"{node.sdfg.name}_{param}{paramval}"
+                    for nstate in node.sdfg.nodes():
+                        backup.extend(
+                            nsdfg_prepare_unroll(nstate, paramname, paramval))
+                    if param in node.symbol_mapping:
+                        node.symbol_mapping.pop(param)
+                    node.sdfg.add_constant(param, int(paramval))
+            return backup
+
+        def nsdfg_after_unroll(backup):
+            for node, unique_name, name, symbols, constants in backup:
+                node.unique_name = unique_name
+                node.sdfg.name = name
+                node.symbol_mapping = symbols
+                node.sdfg.constants_prop = constants
+
         for begin, end, stride in entry_node.map.range:
             l = []
             while begin <= end:
                 l.append(begin)
                 begin += stride
             index_list.append(l)
 
+        sdfgconsts = sdfg.constants_prop
+        sdfg.constants_prop = copy.deepcopy(sdfg.constants_prop)
+
         for indices in product(*index_list):
             callsite_stream.write('{')
+            nsdfg_unroll_info = None
             for param, index in zip(entry_node.map.params, indices):
-                callsite_stream.write(f'auto {param} = {sym2cpp(index)};')
-            self._dispatcher.dispatch_subgraph(sdfg, scope, state_id,
-                                            function_stream, callsite_stream,
-                                            skip_entry_node=True,
-                                            skip_exit_node=True)
+                if nsdfg_unroll_info is None:
+                    nsdfg_unroll_info = nsdfg_prepare_unroll(
+                        scope, str(param), str(index))
+                else:
+                    nsdfg_prepare_unroll(scope, str(param), str(index))
+                callsite_stream.write(
+                    f"constexpr long long {param} = {dace.codegen.targets.common.sym2cpp(index)};\n",
+                    sdfg)
+                sdfg.add_constant(param, int(index))
+
+            callsite_stream.write('{')
```

Comment:
I'm actually not sure, but I don't think it's a problem, because you anyway have to be in a map to use this schedule type, which means you are inside function-code, with the exception of nested sdfg's, but they are handled seperately.

Reply:
In fact I did not even change that, one curly brace was already there, I just added doing a second one